mkosi/_version.py

```diff
@@ -0,0 +1,32 @@
+# SPDX-License-Identifier: LGPL-2.1-or-later
+
+import datetime
+import os
+import subprocess
+from importlib.metadata import PackageNotFoundError, version
+from pathlib import Path
+
+try:
+    __version__ = version("package-name")
+except PackageNotFoundError:
+    # This branch means, that mkosi was not installed as a package as Python is aware, this is most likely
+    # the case if mkosi was run via the bin/mkosi shim in mkosi's repository. Though discouraged, we fall
+    # back to importing setuptools_scm directly and having it try figure stuff out.
+    try:
+        p = subprocess.run(
+            ["git", "describe"],
+            cwd=Path(__file__).parent.parent,
+            check=True,
+            text=True,
+            capture_output=True,
+        )
+        version, *rest = p.stdout.strip().split("-")
+        version = version.lstrip("v")
+        if rest:
+            numcommits, commit = rest
+            __version__ = f"{version}.post1.dev{numcommits}+{commit}.d{datetime.datetime.now():%Y%m%d}"
+        else:
+            # we are exactly on a trag
+            __version__ = version
+    except (subprocess.CalledProcessError, NotADirectoryError):
```

Comment:
Can also fail with `FileNotFoundError` if git is not installed (e.g. someone just downloaded the archive via the browser)